contrib/globalsign/mgo/mgo.go

```diff
@@ -0,0 +1,335 @@
+// Package mgo provides functions and types which allow tracing of the MGO MongoDB client (https://github.com/globalsign/mgo)
+package mgo // import "gopkg.in/DataDog/dd-trace-go.v1/contrib/globalsign/mgo"
+
+import (
+	"github.com/globalsign/mgo"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// Dial opens a connection to a MongoDB server and ties it to the
+// given context for tracing MongoDB calls.
+func Dial(url string, opts ...MongoOption) (*Session, error) {
+	session, err := mgo.Dial(url)
+	s := &Session{
+		Session: session,
+	}
+
+	defaults(&s.cfg)
+	for _, fn := range opts {
+		fn(&s.cfg)
+	}
+
+	return s, err
+}
+
+// Session wraps a Session with the Context and Config information
+// needed to support tracing.
+type Session struct {
+	*mgo.Session
+
+	cfg mongoConfig
```

Comment:
Let's use a pointer when we reference this config all around. No need to make copies of it.